src/sail_sv_backend/jib_sv.ml

```diff
@@ -535,11 +541,16 @@ let collect_spec_info ctx cdefs =
       )
       CT_unit cdefs
   in
+  let global_names : StringSet.t =
+    NameSet.fold (fun name acc -> StringSet.add (string_of_name ~zencode:false name) acc) global_lets StringSet.empty
+  in
+  sv_globals := Some global_names;
```

Comment:
The point of packaging all the information this function gathers into a `spec_info` structure is specifically to avoid the use of global state.